.github/workflows/ci_tests.yml

```diff
@@ -31,39 +31,32 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        platform: [ubuntu-latest]
+        platform: [ubuntu-24.04, macos-15]
         compiler:
-          - cpp: g++
-            c: gcc
-          - cpp: clang++
-            c: clang
+          - cpp: g++-14
+            c: gcc-14
+          - cpp: clang++-18
+            c: clang-18
         cpp_version: [17, 20, 23, 26]
         cmake_args:
           - description: "Default"
             args: ""
-          - description: "TSan"
-            args: "-DCMAKE_CXX_FLAGS=-fsanitize=thread"
-          - description: "ASan"
-            args: "-DCMAKE_CXX_FLAGS='-fsanitize=address -fsanitize=undefined'"
+          # FIXME: find a better solution to be usable for multible OS builds on CI! CK
+          # - description: "TSan"
+          #  args: "-DCMAKE_CXX_FLAGS=-fsanitize=thread"
+          # - description: "ASan"
+          #   args: "-DCMAKE_CXX_FLAGS='-fsanitize=address -fsanitize=undefined'"
```

Comment:
better solution  may be found here: https://github.com/beman-project/execution26/tree/main/.github/workflows

Reply:
Can you provide more description here? What are you trying to do?

Reply:
Hey CK, I would suggest you defer further improvement to another PR. Leaving a note here with some explanation on what improvements would be needed should be sufficient.

This PR is currently already complex and changes suggested to the build system are affecting fundamentals to our setup, it may be best to leave this PR in a stable state so other beman project members can perform reviews. As it may take some time for members to discuss the implications of the build updates.

Having small, atomic PRs also help all of us manage changes easily, as the longer this PR is hanging in the review queue, the more out of date this PR is in respect to other pending improvements, this applies to both sides. 

Again thank you for your contribution.

Reply:
Yes, at the moment, I have no plan to change more on this **PR**.
But there are to **FIXME** in my changes and I have no advice how to continue?

For me, your **CI** workflow tries to match.

I would split the CI in **OS** specific parts, reduce the compiler versions, and the C++ standards you want to check.
Also I would also create **CI** specific `cmake workflow presets` and use them.

see too [CMakeWorkflowPresets: How to keep it as simple as possible?](https://discourse.cmake.org/t/cmakeworkflowpresets-how-to-keep-it-as-simple-as-possible/8735)